express-notes-api/index.js

```diff
@@ -0,0 +1,104 @@
+const express = require('express');
+const app = express();
+const data = require('./data.json');
+const jsonMiddleware = express.json();
+
+app.use(jsonMiddleware);
+const fs = require('fs');
+
+app.get('/api/notes', (req, res) => {
+  const newArr = [];
+  for (const x in data.notes) {
+    newArr.push(data.notes[x]);
+  }
+  res.json(newArr);
+});
+app.get('/api/notes/:id', (req, res) => {
+  const id = Number(req.params.id);
+  if (id > 0) {
+    if (!data.notes[id]) {
+      res.status(404).send({ error: `cannot find note with id ${id}` });
+    } else {
+      res.status(200).json(data.notes[id]);
+    }
+  } else if (id < 0) {
+    res.status(400).json({ error: 'id must be a positive integer' });
+  }
+});
+
+app.post('/api/notes', (req, res) => {
+  if (req.body.content === undefined) {
+    res.status(400).json({ error: 'content is a required field' });
+
+  } else if (req.body.content !== undefined) {
```

Comment:
Just an `else` here is better, you already checked `req.body.content`, no point to check the opposite of what was already checked.